common/domain/dlq_message_handler_test.go

```diff
@@ -102,6 +110,146 @@ func (s *dlqMessageHandlerSuite) TestReadMessages() {
 	s.Nil(token)
 }
 
+func TestDLQMessageHandler_Start(t *testing.T) {
+	tests := []struct {
+		name           string
+		setupMocks     func(m *MockReplicationQueue)
+		initialStatus  int32
+		expectedStatus int32
+	}{
+		{
+			name: "Should start when initialized",
+			setupMocks: func(m *MockReplicationQueue) {
+				m.EXPECT().GetDLQSize(gomock.Any()).Return(int64(1), nil).AnyTimes()
+			},
+			initialStatus:  common.DaemonStatusInitialized,
+			expectedStatus: common.DaemonStatusStarted,
+		},
+		{
+			name: "Should not start when already started",
+			setupMocks: func(m *MockReplicationQueue) {
+				// No calls expected since the handler should recognize it's already started
+			},
+			initialStatus:  common.DaemonStatusStarted,
+			expectedStatus: common.DaemonStatusStarted,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
```

Comment:
```suggestion
```
gomock has done this automatically via `t.Cleanup` for 4 years now: https://github.com/golang/mock/issues/407